network/src/peers/peers.rs

```diff
@@ -247,12 +250,13 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
 
     /// Broadcasts a `GetPeers` message to all connected peers to request for more peers.
     async fn broadcast_getpeers_requests(&self) {
-        // Check that this node is not a bootnode.
-        if !self.config.is_bootnode() {
+        // If the node is not a bootnode or a crawler, check if the request for peers is needed
+        // based on the number of active connections.
+        if self.config.is_regular_node() {
```

Comment:
this is the check that accidentally omitted (and effectively included) crawlers when they were separated from bootnodes